### PR TITLE
Update container

### PR DESCRIPTION
**Pull request by tabic**
I changed the following:

1. Use build and app stage
    We don't need node and all the other things from the build in the final image
2. Node 11 is EOL, therefore we are back on node 10. Node 12 is not fully supported yet
3. Add CI

### CODE REVIEW

**Comment by tabic** [comment]
The build fails due to an upstream issue, https://github.com/gchq/CyberChef/issues/1162. The patches should work but require testing once upstream is fixed.

**Comment by tabic** [comment]
@mpepping back then you released new versions in a separate branches. I think we should add an build ARG VERSION which defaults to master and gets overwritten by GitHub actions.
I am not sure if DockerHub auto build feature can build different versions and also re-build master automatically, therefore I suggest switching to GitHub actions for pushing the image.

**Comment by mpepping** [comment]
@rxcket Looks good! The branches are for image tags indeed. That's how tags initially worked on [Docker Hub automated builds](https://docs.docker.com/docker-hub/builds/#tag-and-branch-builds). An `ARG VERSION` var is a good idea. A good way to get rid of all those branches. 
+1 for the GitHub CI switch. I've added CI token to this repo. 

Building your PR branch throws the `Module not found errors`. I'll subscribe on the upstream issue.

**Comment by tabic** [comment]
The "bug" is fixed, npm install does not run scripts if executed as root. Container is working as expected.

Can you please tell me the var names for dockerhub login or add it yourself? See https://github.com/marketplace/actions/build-and-push-docker-images#git-context

**Comment by mpepping** [comment]
@rxcket Nice work! We're using the defaults: `${{ secrets.DOCKERHUB_USERNAME }}` and `${{ secrets.DOCKERHUB_TOKEN }}`

**Review by mpepping — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      -
+        name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
+
+      -
+        name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+
```
Comment (mpepping) [inline .github/workflows/ci.yml:24]:
```suggestion
      -
        name: Login to DockerHub
        uses: docker/login-action@v1 
        with:
          username: ${{ secrets.DOCKERHUB_USERNAME }}
          password: ${{ secrets.DOCKERHUB_TOKEN }}

```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+      -
+        name: Build
+        uses: docker/build-push-action@v2
+        with:
+          tags: mpepping/cyberchef:latest
```
Comment (mpepping) [inline .github/workflows/ci.yml]:
```suggestion
          push: true
          tags: mpepping/cyberchef:latest
```

**Review by mpepping — APPROVED** [review]

**Comment by tabic** [comment]
Added release and master build + daily rebuilds

**Comment by mpepping** [comment]
Thanks for the contribution @rxcket! LGTM! 
I'll tune the rebuild schedule depending on how many GitHub action minutes building images for all platforms will 'cost'.

**Pull request merged**